src/theory/strings/theory_strings.cpp

```diff
@@ -1147,8 +1251,40 @@ void TheoryStrings::eqNotifyNewClass(TNode t){
     }
     //we care about the length of this string
     registerTerm( t[0], 1 );
-  }else{
-    //getExtTheory()->registerTerm( t );
+    return;
+  }
+  else if (k == CONST_STRING)
+  {
+    EqcInfo* ei = getOrMakeEqcInfo(t);
+    ei->d_prefixC = t;
+    ei->d_suffixC = t;
+    return;
+  }
+  else if (k == STRING_CONCAT)
+  {
+    addPrefixToEqcInfo(t, t, t);
+  }
+}
+
+void TheoryStrings::addPrefixToEqcInfo(Node t, Node concat, Node eqc)
+{
+  EqcInfo* ei = nullptr;
+  // check each side
+  for (unsigned r = 0; r < 2; r++)
+  {
+    unsigned index = r == 0 ? 0 : concat.getNumChildren() - 1;
+    Node c = utils::getConstantComponent(concat[index]);
```

Comment:
Aren't the two lines above essentially what `getConstantPrefix()` does?

Reply:
Yes; this was done in a more verbose way for optimization.
In my estimation, the performance of this code is highly critical (it is run at standard effort), so this is e.g. ensuring we don't have to check the kind twice.

Reply:
I don't think that it matters too much but fair enough :)